Backend dev env setup docs

limivann opened the pull request:
[SCSE-22]

limivann committed:
New commit [SCSE-22]

ztjhz committed:
docs: add python installation instructions

This commit adds the python installation instruction for
windows, macos, and linux to the backend dev environment setup page.

[SCSE-22]

ztjhz committed:
docs: add peotry instruction + restructure docs

This commits adds the peotry instruction for windows, macos, and linus
to the backend dev environment setup page.
It also restructures the backend dev environment setup page.

[SCSE-22]

ztjhz committed:
docs: add serverless cli and dynamodb offline

This commit adds documentation for installing serverless cli and
dynamodb offline

[SCSE-22]

RealDyllon reviewed (changes requested):
Looks good, just needs 2 changes:

1. rename all instances (in file names and file content) of `peotry` to `poetry`

2. Instead of running dynamodb inside java locally, let's use this: https://www.serverless.com/plugins/serverless-dynamodb-local

ztjhz committed:
docs: fix typo from peotry to poetry

This commit renames all instances (in file names and file content) of
peotry to poetry
[SCSE-22]

ztjhz committed:
docs: update DynamoDB installation instruction

This commit updates the DynamoDB installation instruction from
running it locally inside Java to running it with
the serverless-dynamodb-local plugin
[SCSE-22]
